### PR TITLE
Update matplotlib35

### LINKED ISSUE #2059: [TASK] Update matplotlib to at least 3.5
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
Matplotlib has deprecation warnings when run with newer pythons: https://github.com/gmlc-dispatches/dispatches/issues/173

**Describe the solution you'd like**
Update matplotlib to a newer version.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #2059 

##### What are the significant changes in functionality due to this change request?
Updates matplotlib to 3.5 and fixes various deprecations.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: rook/Tester.py [hunk 1]
```diff
@@ -24,7 +24,6 @@
 import time
 import threading
 import platform
-from distutils import spawn
```
Comment (joshua-cogliati-inl) [inline rook/Tester.py:27]: Distutils is deprecated, so remove

File: ravenframework/GridEntities.py [hunk 2]
```diff
@@ -504,9 +504,9 @@ def initialize(self, initDictionary=None):
       if self.gridContainer['transformationMethods'] is not None:
         if varName in self.gridContainer['transformationMethods']:
           self.gridContainer['gridVectors'][varName] = np.asarray([self.gridContainer['transformationMethods'][varName][0](coor) for coor in self.gridContainer['gridVectors'][varName]])
-      pointByVar[varId]                              = np.int(np.shape(self.gridContainer['gridVectors'][varName])[0])
```
Comment (joshua-cogliati-inl) [inline ravenframework/GridEntities.py:507]: np.int is deprecated so remove

File: ravenframework/Models/PostProcessors/LimitSurfaceIntegral.py [hunk 3]
```diff
@@ -253,7 +253,7 @@ def run(self, input):
         if self.variableDist[varName] == None:
           randomMatrix[:, index] = randomMatrix[:, index] * (self.lowerUpperDict[varName]['upperBound'] - self.lowerUpperDict[varName]['lowerBound']) + self.lowerUpperDict[varName]['lowerBound']
         else:
-          f = np.vectorize(self.variableDist[varName].ppf, otypes=[np.float])
```
Comment (joshua-cogliati-inl) [inline ravenframework/Models/PostProcessors/LimitSurfaceIntegral.py:256]: np.float is deprecated so remove

File: ravenframework/OutStreams/PlotInterfaces/GeneralPlot.py [hunk 4]
```diff
@@ -620,10 +620,23 @@ def __executeActions(self):
       elif key == 'title':
         self.ax.set_title(self.options[key]['text'], **self.options[key].get('attributes', {}))
       elif key == 'scale':
+        major, minor = [int(x) for x in matplotlib.__version__.split('.')[:2]]
+        #matplotlib before 3.5 used nonpos instead of nonpositive
```
Comment (joshua-cogliati-inl) [inline ravenframework/OutStreams/PlotInterfaces/GeneralPlot.py:624]: nonposx and nonposy have been replaced with nonpositive in matplotlib 3.5, but have also been removed for anything other than a log scale.

**Review by joshua-cogliati-inl — COMMENTED** [review]
Adding comments on reasons for changes.

**Comment by joshua-cogliati-inl** [comment]
FYI: The Fedora 32 test passed, as can be checked by following the link: https://civet.inl.gov/job/1334464/

**Review by wangcj05 — APPROVED** [review]
Changes are good.

**Comment by wangcj05** [comment]
libraries update, and resolve some issues corresponding the update. PR looks good, and checklist is satisfied, PR can be merged.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl Could you draft an email to the user about the library update?

**Pull request merged**